pkg/clients/connection/connection.go

```diff
@@ -82,9 +82,9 @@ type Observation struct {
 
 // UpdateStatus updates any fields of the supplied ConnectionStatus to
 // reflect the state of the supplied Address.
-func UpdateStatus(s *v1alpha3.ConnectionStatus, o Observation) {
-	s.Peering = o.Connection.Peering
-	s.Service = o.Connection.Service
+func UpdateStatus(s *v1beta1.ConnectionStatus, o Observation) {
```

Comment:
I don't really love that this is in the `client` package, but I also don't think it really fits in `controller`. Any thoughts on this @negz? We could set just return the peering state and then actually set the condition in `Observe`...

Reply:
No strong feelings either way on this one. I wrote this controller, so I must have felt like the client package was appropriate at the time. 🤔